lib/analyzer/package-managers/apt.ts

```diff
@@ -43,25 +44,53 @@ export function analyzeDistroless(
 
 function parseDpkgFile(text: string): AnalyzedPackageWithVersion[] {
   const pkgs: AnalyzedPackageWithVersion[] = [];
-  let curPkg: any = null;
+  let curPkg: AnalyzedPackageWithVersion | null = null;
   for (const line of text.split("\n")) {
-    curPkg = parseDpkgLine(line, curPkg, pkgs);
+    curPkg = parseDpkgLine(line, curPkg!, pkgs);
+    assignPurl(curPkg);
```

Comment:
I think it would be more explicit to instead "compute" a purl and return it, e.g.
```
curPkg.Purl = purl(curPkg)
```